link.go

```diff
@@ -539,11 +573,48 @@ func (l *link) muxReceive(fr performTransfer) error {
 
 	// decrement link-credit after entire message received
 	l.deliveryCount++
-	l.linkCredit--
+
+	atomic.AddUint32(&l.linkCredit, ^uint32(0))
+	//l.linkCredit--
```

Comment:
Remove?

Reply:
All atomic calls removed.

Reply:
(well, and comment restored to being the actual code).